archivist/dictmerge.py

```diff
@@ -28,3 +28,44 @@ def _dotdict(dct: Optional[dict]) -> Optional[dict]:
     if dct is None:
         return None
     return flatten(dct, reducer="dot")
+
+
+def assets_ext_attr(assets: list) -> list:
+    """Create list of assets with extended attribute(s)"""
+    extended_attributes_asset = [
+        asset
+        for asset in assets
+        if any(not item.startswith("arc_") for item in asset["attributes"])
+    ]
+    return extended_attributes_asset
+
+
+def attachment_identities_assets(assets_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    for asset in assets_with_attachments:
+        total_attachments_assets = {
+            item["arc_attachment_identity"]
+            for item in asset["attributes"]["arc_attachments"]
+        }
+    total_attachments_assets = list(filter(None, total_attachments_assets))
+    return total_attachments_assets
+
+
+def events_ext_attr(events: list) -> list:
+    """Create list of events with extended attribute(s)"""
+    extended_attributes_event = [
+        event
+        for event in events
+        if any(not item.startswith("arc_") for item in event["event_attributes"])
+    ]
+    return extended_attributes_event
+
+
+def attachment_identities_events(events_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    for event in events_with_attachments:
+        total_attachments_events = {
+            item["arc_attachment_identity"]
+            for item in event["event_attributes"]["arc_attachments"]
+        }
```

Comment:
This is incorrect - the returned value will only contain identities from the last event as the list is redone every time.
also this return a set and not a list (curly braces constitute a set)

    total_attachments_events = {
         item["arc_attachment_identity"]
         for event in events_with_attachments
         for item in event["event_attributes"]["arc_attachments"]
    }

Reply:
The set was used so there would be no duplicates, I had issues with duplication in previous versions of the code. Is the list redone because of the preceding for loop?

Reply:
This method has been changed to now filter assets with an "arc_attachments" attribute, so the set/loop is no longer required and has been removed.